### PR TITLE
Prop names of JSON used in validation /6

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #26566
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-7.0&branch=pr-en-us-26567#use-json-property-names-in-validation-errors)

### CODE REVIEW

File: aspnetcore/mvc/models/validation.md [hunk 1]
```diff
@@ -467,6 +467,23 @@ When applied to a `Name` property, the error message created by the preceding co
 
 To find out which parameters are passed to `String.Format` for a particular attribute's error message, see the [DataAnnotations source code](https://github.com/dotnet/runtime/tree/main/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations).
 
+### Use JSON property names in validation errors
+
+By default, when an validation error occurs, model validation produces a <xref:Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary> with the property name as the error key. Some apps, such as single page apps, benefit from using JSON property names for validation errors. The following code configures validation to use the [`SystemTextJsonValidationMetadataProvider`](/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.metadata.systemtextjsonvalidationmetadataprovider) to use JSON property names:
+
+:::code language="csharp" source="~/mvc/models/validation/samples/7.x/ValidationJSON/Program.cs" id="snippet_1" highlight="5-8":::
+
+The following code sets the policy to use camel-casing:
```
Comment (brunolins16) [inline aspnetcore/mvc/models/validation.md:476]: What do you think about keep the doc simple and remove the camel-casing example for both STJ and NewtonSoft?

File: aspnetcore/mvc/models/validation.md [hunk 2]
```diff
@@ -467,6 +467,23 @@ When applied to a `Name` property, the error message created by the preceding co
 
 To find out which parameters are passed to `String.Format` for a particular attribute's error message, see the [DataAnnotations source code](https://github.com/dotnet/runtime/tree/main/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations).
 
+### Use JSON property names in validation errors
+
+By default, when an validation error occurs, model validation produces a <xref:Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary> with the property name as the error key. Some apps, such as single page apps, benefit from using JSON property names for validation errors. The following code configures validation to use the [`SystemTextJsonValidationMetadataProvider`](/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.metadata.systemtextjsonvalidationmetadataprovider) to use JSON property names:
```
Comment (brunolins16) [inline aspnetcore/mvc/models/validation.md:472]: The api doc (/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.metadata.systemtextjsonvalidationmetadataprovider) will eventually get available, right?

Comment (brunolins16) [inline aspnetcore/mvc/models/validation.md:472]:
```suggestion
By default, when an validation error occurs, model validation produces a <xref:Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary> with the property name as the error key. Some apps, such as single page apps, benefit from using JSON property names for validation errors generated from Web APIs. The following code configures validation to use the [`SystemTextJsonValidationMetadataProvider`](/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.metadata.systemtextjsonvalidationmetadataprovider) to use JSON property names:
```

What do you think about mentioning "Web APIs"?

Comment (Rick-Anderson) [reply aspnetcore/mvc/models/validation.md]: The links work now. You just have to remove `.review` in the URL when you click the link.

`https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.metadata.systemtextjsonvalidationmetadataprovider`
The links will work in the published doc.

**Review by brunolins16 — APPROVED** [review]

**Pull request merged**